actionpack/lib/action_controller/metal/strong_parameters.rb

```diff
@@ -646,9 +656,18 @@ def require(key)
     #     params.permit(person: { '0': [:email], '1': [:phone]}).to_h
     #     # => {"person"=>{"0"=>{"email"=>"none@test.com"}, "1"=>{"phone"=>"555-6789"}}}
     def permit(*filters)
-      params = self.class.new
+      permit_filters(filters, &method(:unpermitted_parameters))
+    end
+
+    # Same as permit, but raises on unpermitted parameters by default.
+    # Ignores ActionController::Parameters.action_on_unpermitted_parameters
+    #
+    # Accepts a block which will yield any unpermitted keys instead of raising.
+    def permit!(*filters, &block)
```

Comment:
We already have a `permit!` method.

Reply:
uh oh, haha. Of course. When we were talking about `expect!` in the other PR, I completely forgot about this. Hmmmm...